src/Tools/dotnet-monitor/Auth/ApiKeyAuthenticationHandler.cs

```diff
@@ -24,7 +24,10 @@ namespace Microsoft.Diagnostics.Tools.Monitor
     /// </summary>
     internal sealed class ApiKeyAuthenticationHandler : AuthenticationHandler<ApiKeyAuthenticationOptions>
     {
-        public const int ApiKeyNumBytes = 32;
+        public const int ApiKeyByteMinLength = 32;
+        public const int ApiKeyByteMaxLength = 2048;
+        // This is the max length to efficiently encode an ApiKey of the length ApiKeyByteMaxLength.
+        private const int ApiKeyBase64MaxLength = ((ApiKeyByteMaxLength * 8) / 6) + 2;
```

Comment:
This doesn't seem quite right.

For example, 3 bytes encoded into base64 requires 4 characters. The calculation here says (3 * 8) / 6 + 2 = 6 characters.
| Byte Count | Encoded Length |
|---|---|
| 0 | 2 |
| 1 | 3 |
| 2 | 4 |
| 3 | 6 |
| 4 | 7 |

I think the algorithm is (byteLength + 2) / 3 * 4 giving:
| Byte Count | Encoded Length |
|---|---|
| 0 | 0 |
| 1 | 4 |
| 2 | 4 |
| 3 | 4 |
| 4 | 8 |

I think you can use System.Buffers.Text.Base64.GetMaxEncodedToUtf8Length to calculate the static value and not have to think about the algorithm :-)

Reply:
I wish I knew about `System.Buffers.Text.Base64.GetMaxEncodedToUtf8Length` before doing this. I couldn't find any of this.

Reply:
Done